.github/workflows/master.yml

```diff
@@ -269,14 +257,256 @@ jobs:
+        python-version: [3.9.x]
+        dotnet: [3.1.x]
+    runs-on: windows-latest
+    needs: windows-build
+    env:
+      GOPATH: ${{ github.workspace }}
+      PULUMI_LOCAL_NUGET: "D:\\Pulumi\\nuget"
+      ACTIONS_ALLOW_UNSECURE_COMMANDS: true
+    steps:
+      - name: Install DotNet ${{ matrix.dotnet }}
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ matrix.dotnet }}
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Set up Go ${{ matrix.go-version }}
+        uses: actions/setup-go@v1
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install Python Deps
+        run: |
+          pip3 install pyenv-win
+          pip3 install pipenv
+      - name: Ensure Pulumi Bin exists
+        run: mkdir D:\Pulumi\bin
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v2
+        with:
+          path: ./src/github.com/${{ github.repository }}
+      - name: Fetch Tags
+        run: |
+          cd ./src/github.com/${{ github.repository }} && git fetch --quiet --prune --unshallow --tags
+      - name: Ensure Local Nuget Exists
+        run: mkdir D:/pulumi/nuget
+      - name: Download Nuget Archive
+        uses: actions/download-artifact@v2
+        with:
+          name: pulumi-nuget.tar.gz
+          path: ${{ github.workspace}}
+      - name: Uncompress Nuget folder
+        run: tar -zxf ${{ github.workspace}}/nuget.tar.gz -C D:\Pulumi\nuget
+      - name: Download Pulumi CLI Archive
+        uses: actions/download-artifact@v2
+        with:
+          name: pulumi-windows-cli.tar.gz
+          path: ${{ github.workspace}}
+      - name: Uncompress CLI folder
+        run: |
+          tar -zxf ${{ github.workspace}}/cli.tar.gz -C D:\Pulumi\bin
+      - run: dir D:\Pulumi\bin
+      - name: Set Build Env Vars
+        shell: bash
+        run: |
+          echo "D:\Pulumi\bin" >> $GITHUB_PATH
+      - name: Download Pulumi NodeJS Archive
+        uses: actions/download-artifact@v2
+        with:
+          name: pulumi-windows-nodejs-sdk.tar.gz
+          path: ${{ github.workspace}}\src\github.com\${{ github.repository }}
+      - name: Download Pulumi Python Archive
+        uses: actions/download-artifact@v2
+        with:
+          name: pulumi-windows-python-sdk.tar.gz
+          path: ${{ github.workspace}}\src\github.com\${{ github.repository }}
+      - name: Uncompress SDK folder
+        run: |
+          cd src\github.com\${{ github.repository }}
+          tar -zxf ./nodejs-sdk.tar.gz -C sdk\nodejs
+          tar -zxf ./python-sdk.tar.gz -C sdk\python
+      - name: Uninstall Choco Pulumi
+        run: choco uninstall pulumi -y
```

Comment:
Why do we do this? It looks like this step took 3 minutes in the action, wondering if there's a way to avoid it.

Reply:
This is a temp step that I am trying to work through a way to remove - this is ensuring that the version of Pulumi we have is the *exact* version we want